---
## **`APM1111 - FA 6`**
#### **Author: Sean Angelo T. Lagua**
#### **Date: November 11, 2023**
#### **Output: Interpreting t-test**
---

Given an enclosed community of 24 people in a distant future where technological advancement has led to the discovery of invisible cloaks, we examine the behaviour of these participants in the way they perform mischief. With the given data we can extract meaniningful interpretations concerning the behaviour of people from the future.

In [91]:
import pandas as pd
import numpy as np
import scipy
import math

# Reading the csv file and turning it into a dataframe
data = pd.read_csv("Invisibility Cloak.csv")
data_frame = pd.DataFrame(data)

#Separating the group of people with cloak and no cloak
no_cloak = []
cloak = []
for index, row in data_frame.iterrows() :
    if(row['Cloak'] == 0):
        no_cloak.append([row['Cloak'], row['Mischief']])
    else:
        cloak.append([row['Cloak'], row['Mischief']])

no_cloak_df = pd.DataFrame(no_cloak, columns=["Cloak", "Mischief"])
cloak_df = pd.DataFrame(cloak, columns=["Cloak", "Mischief"])

#Getting the summary of the each group's statistic
no_cloak_mean = np.mean(no_cloak_df['Mischief'])
no_cloak_std = np.std(no_cloak_df['Mischief'], ddof=1)
no_cloak_stderror = no_cloak_std / math.sqrt(len(no_cloak))
no_cloak_var = no_cloak_std**2
cloak_mean = np.mean(cloak_df['Mischief'])
cloak_std = np.std(cloak_df['Mischief'], ddof=1)
cloak_stderror = cloak_std / math.sqrt(len(cloak))
cloak_var = cloak_std**2

#Presenting the data extracted into a table
independent_data = {
                    'Mischief': ['No Invisibility Cloak', 'Invisibility Cloak'],
                    'Mean': [no_cloak_mean, cloak_mean],
                    'N': [len(no_cloak), len(cloak)],
                    'Std. Deviation': [no_cloak_std, cloak_std],
                    'Std. Error Mean': [no_cloak_stderror, cloak_stderror]
                   }

ind_df = pd.DataFrame(independent_data)
ind_df

Mischief  Mean   N  Std. Deviation  Std. Error Mean
0  No Invisibility Cloak  3.75  12        1.912875         0.552199
1     Invisibility Cloak  5.00  12        1.651446         0.476731

From the table above, we can see a summary of the statistics for the two conditions given. We can notice that both groups had 12 participants labelled by $N$. With the given mean, we can say that the group that does not have an invisibility cloak performed less mischievous acts than the group that had the invisibility cloak. Moreover, the group with no invisibility cloak has 0.552 while the other group has 0.477. Now, we move on to getting the main test statistic for both groups.

In [111]:
df = len(no_cloak_df['Mischief']) + len(cloak_df['Mischief']) - 2
Sp_num = (no_cloak_var*(len(no_cloak_df['Mischief']) - 1)) + (cloak_var*(len(cloak_df['Mischief']) - 1))
Sp = math.sqrt(Sp_num / df)
pos_CI = (no_cloak_mean - cloak_mean) + 1.717*math.sqrt(((Sp**2)/len(no_cloak_df['Mischief'])) + ((Sp**2)/len(cloak_df['Mischief'])))
neg_CI = (no_cloak_mean - cloak_mean) - 1.717*math.sqrt(((Sp**2)/len(no_cloak_df['Mischief'])) + ((Sp**2)/len(cloak_df['Mischief'])))
cohen_d = (no_cloak_mean - cloak_mean) / Sp

t_value_ind = (no_cloak_mean - cloak_mean) / math.sqrt((no_cloak_var/ len(no_cloak_df['Mischief'])) + (cloak_var/ len(cloak_df['Mischief'])))

print(f"The calculated t-score is t({df}) = {round(t_value_ind, 5)} with p = .101, d = {round(cohen_d, 5)} 95% CI [{round(neg_CI, 5)}, {round(pos_CI, 5)}]")

The calculated t-score is t(22) = -1.71346 with p = .101, d = -0.69952 95% CI [-2.50258, 0.00258]


**Presenting the Results**

In a distant future with superb technological advancement, a group of 24 participants were divided equally into two and placed in an enclosed community. Hidden cameras were placed around the community to record their mischievous acts; half were given an invisibility cloak. In a span of a week, the gathered reveals that those who were given the invisibility cloak performed significantly more mischief ($M =$ 5, $SD =$ 1.651, $n =$ 12) than those who were not given an invisibility cloak ($M =$ 3.75, $SD =$ 1.913, $n =$ 12), $t$(22) $=$ -1.71346, $p <$ .101, d = 0.69, 95% CI [-2.50258, 0.00258$].

In [118]:
mischief_pair = []
for index in range(int((len(no_cloak_df['Mischief']) + len(cloak_df['Mischief'])) / 2)):
    if(no_cloak[index][0] == 0 and cloak[index][0] == 1):
        mischief_pair.append([no_cloak[index][1], cloak[index][1]])

mschf_pair_diff = []
for j in range(len(mischief_pair)):
    mschf_pair_diff.append(mischief_pair[j][0] - mischief_pair[j][1])

df = len(mischief_pair) - 1
mschf_pair_mean = np.mean(mschf_pair_diff)
mschf_pair_stdev = np.std(mschf_pair_diff, ddof=1)
mschf_pair_stderror = mschf_pair_stdev / math.sqrt(len(mischief_pair))

# -2.201 -> Critical T-Value
t_value_pair = (mschf_pair_mean - 0) / mschf_pair_stderror
#Reject null hypotheses

print(f"The calculated t-value is: {t_value_pair}.")

The calculated t-value is: -3.8044295512634103.


We can see that the calculated t-score is less than the critical t-value of -2.201, thus, we reject the null hypotheses and say that there is a significant statistical difference in the behaviour of the people in the experiment.

**Presenting the Results**

Reporting now the same information but as a matched-sample t-test, we see that when given the invisibility cloak, the participants performed more mischievous acts ($M =$ 1.25, $SD =$ 1.138, $SE =$ 0.32856), $t$(11) = -3.80443, p = .003.